packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -168,6 +168,12 @@ abstract class FlutterCommand extends Command<void> {
         'will select a random open port on the host.',
       hide: hide,
     );
+    argParser.addFlag('web-force-ws',
```

Comment:
How about `web-enable-sse` defaulting to `true`, with the option to negate it via `--no-web-enable-sse`?

I would also add some context to when this might be necessary. Maybe certain hosts block sse connections?

Reply:
`--no-web-enable-sse` sounds a little like it's enabling/disabling a feature rather than selecting between the protocol, would `--web-use-sse` make more sense? (I'm happy to go with whatever - the name isn't so important to me, it's for the extension to provide automatically).

> I would also add some context to when this might be necessary. Maybe certain hosts block sse connections?

Heh, I did have something but removed it for being wordy. Does something like "This is used by editors that are unable to use SSE for connecting to the debug service proxy." make sense? It's more about the debugger client not supporting SSE than it being blocked - since the SSE implementation in Dart is bi-directional (which is different to just plain old SSE) and non-Dart clients would need to re-implement the HTTP POST side of things.

Reply:
If its selecting a protocol, I would make it something like `--web-debug-protocol=sse/ws`, you can specify allowed and default values too

Reply:
I've changed it to this. It's a bit weird that it's only for the web-server device though (Chrome always uses WebSockets), so I put this in the description. I'm wondering if `--web-server-debug-protocol` would be more explicit?